### PR TITLE
refactor:accordion item styles

### PR DESCRIPTION
**Pull request by waynewee**
- took some styles out of the accordion item to allow for more customizability

### CODE REVIEW

File: src/components/UI/Accordion/AccordionItem.tsx [hunk 1]
```diff
@@ -1,27 +1,32 @@
 import React, { FunctionComponent, useState } from "react";
+import { addClassNameIfExist } from "../../../utils";
 
 type headingTag = "h3" | "h5";
 
 interface AccordionItemProps {
-  className?: string;
+  containerClassName?: string;
+  collapseClassName?: string;
+  contentClassName?: string;
```
Comment (simboonlong) [inline src/components/UI/Accordion/AccordionItem.tsx]:
```suggestion
  classNameContainer?: string;
  classNameCollapse?: string;
  classNameContent?: string;
```

File: src/components/UI/Accordion/AccordionItem.tsx [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
+  contentClassName,
   headingTag = "h5",
   heading,
   children,
 }) => {
   const [open, setOpen] = useState(false);
   return (
-    <div className={`transition-color duration-200 mb-2${className ? ` ${className}` : ""}`}>
+    <div className={`transition-color duration-200${addClassNameIfExist(containerClassName, { prefix: " " })}`}>
       <div
```
Comment (simboonlong) [inline src/components/UI/Accordion/AccordionItem.tsx]:
```suggestion
    <div className={`transition-color duration-200 ${addClassNameIfExist(containerClassName)}`}>
      <div
```

File: src/components/UI/Accordion/AccordionItem.tsx [hunk 3]
```diff
@@ -1,27 +1,32 @@
...
-        className={`flex justify-between items-center p-4 rounded transition-colors duration-200 cursor-pointer hover:text-cerulean${
+        className={`flex justify-between items-center transition-colors duration-200 cursor-pointer hover:text-cerulean${
           open ? " text-cerulean" : ""
-        }`}
+        }${addClassNameIfExist(collapseClassName, { prefix: " " })}`}
         onClick={() => setOpen(!open)}
```
Comment (simboonlong) [inline src/components/UI/Accordion/AccordionItem.tsx]:
```suggestion
        } ${addClassNameIfExist(collapseClassName)}`}
        onClick={() => setOpen(!open)}
```

File: src/utils/index.ts [hunk 4]
```diff
@@ -45,3 +45,11 @@ export const encode: any = (data: { [x: string]: string | number | boolean }) =>
     .map((key) => encodeURIComponent(key) + "=" + encodeURIComponent(data[key]))
     .join("&");
 };
+
+export const addClassNameIfExist = (className?: string, options?: { prefix?: string; postfix?: string }): string => {
```
Comment (simboonlong) [inline src/utils/index.ts]: add unit tests

File: src/utils/index.ts [hunk 5]
```diff
@@ -45,3 +45,11 @@ export const encode: any = (data: { [x: string]: string | number | boolean }) =>
...
+  if (!className) {
+    return "";
+  }
+
+  return (options?.prefix || "") + className + (options?.postfix || "");
```
Comment (simboonlong) [inline src/utils/index.ts]:
```suggestion

  const prefix = `${options?.prefix ? `${options?.prefix}-` : ""}`;
  const postfix = `${options?.postfix ? `-${options?.postfix}` : ""}`;

  return `${prefix}${className}${postfix}`;
```

File: src/components/UI/Accordion/AccordionItem.tsx (hunk 2 above)
Comment (waynewee) [reply src/components/UI/Accordion/AccordionItem.tsx]: if it doesnt exist the string will be `transition-color duration-200 ` -> whitespace here if u remove the prefix, but can la

**Review by simboonlong — APPROVED** [review]

**Pull request merged**